WordPress/Classes/ViewRelated/Gutenberg/EditorMediaUtility.swift

```diff
@@ -31,8 +31,22 @@ final class AuthenticatedImageDownload: AsyncOperation {
                     throw DownloadError.blogNotFound
                 }
 
-                let blog = try context.existingObject(with: self.blogObjectID) as! Blog
-                return MediaHost(with: blog) { error in
+                var blog: Result<Blog, Error> = .failure(DownloadError.blogNotFound)
```

Comment:
Is this default value necessary? I don't see a code path where it's used, since if a `try` in the `do-catch` throws, the method rethrows.

Reply:
Gotcha. I was hoping we could do something like

```swift
                let x = true
                var foo: Int
                if x {
                    foo = 2
                } else {
                    foo = 3
                }
```

Alas, the construct work in the context of an `if else` but not in a closure

![image](https://github.com/wordpress-mobile/WordPress-iOS/assets/1218433/e59f4ba0-2886-446d-b695-41dc3acd5daf)